cpp/tests/reshape/byte_cast_tests.cpp

```diff
@@ -360,3 +328,34 @@ TEST_F(ByteCastTest, StringValues)
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(output_strings->view(), strings_expected);
 }
+
+TEST_F(ByteCastTest, StringValuesWithNulls)
+{
+  auto const strings_col = [] {
+    auto output =
+      cudf::test::strings_column_wrapper(
+        {"", "The quick", " brown fox...", "!\"#$%&\'()*+,-./", "0123456789:;<=>?@", "[\\]^_`{|}~"})
+        .release();
+
+    // Set nulls by `set_null_mask` so the output column will have non-empty nulls.
+    // This is intentional.
+    auto const null_iter = cudf::test::iterators::nulls_at({2, 4});
+    output->set_null_mask(
+      cudf::test::detail::make_null_mask(null_iter, null_iter + output->size()));
```

Comment:
```suggestion
      cudf::test::detail::make_null_mask(null_iter, null_iter + output->size()), 2);
```
Set the null count too instead of relying on the default.